### DISCUSSION TITLE
Changing interface name based on assigned device via API

### DISCUSSION THREAD

**Discussion by brianatlarge** [netbox-community/netbox#12716] [Help Wanted!] [answered]
I'm wanting to change the names of interfaces to a switch that's added via the API. I can't seem to craft an API call to /api/dcim/interfaces/ that will only give me the interfaces associated with a device. I'd hope I could just do /api/dcim/interfaces/device=whatever but that's not valid.

How are you guys bulk renaming interfaces via the API for a particular device?

**Comment by mtinberg** [comment, marked as answer]
That does work the way you specified, I think you just forgot to make it a query parameter ie

/api/dcim/interfaces/?device=device_name  or ?device_id=id_number


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
From: brianatlarge ***@***.***>
Sent: Thursday, May 25, 2023 9:58 AM
To: netbox-community/netbox ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [netbox-community/netbox] Changing interface name based on assigned device via API (Discussion #12716)


I'm wanting to change the names of interfaces to a switch that's added via the API. I can't seem to craft an API call to /api/dcim/interfaces/ that will only give me the interfaces associated with a device. I'd hope I could just do /api/dcim/interfaces/device=whatever but that's not valid.

How are you guys bulk renaming interfaces via the API for a particular device?

**Reply by brianatlarge** [reply]
Nice, that does exactly what I need. Thanks!
